### ISSUE TITLE
Support for SQLite's fts5 `MATCH` statement

### ISSUE THREAD

**Issue by asm0dey** [jOOQ/jOOQ#13716]
Somehow related to #11172

### Use case:

Now there is an `fts5` type in SQLIte, which allows for the following syntax:

```sql
CREATE VIRTUAL TABLE email USING fts5(sender, title, body);
SELECT * FROM email WHERE email MATCH 'fts5' ORDER BY rank;
```

This example is directly taken from [the docs](https://www.sqlite.org/fts5.html)

Interesting thing here is that we use the whole table name as part of predicate before the `MATCH` keyword.

Please describe the use-case you have. This will better help us understand the context in which you're looking for a new feature.

### Possible workarounds:

Indeed we can just enumerate all columns in the predicate

**Comment by lukaseder** [comment]
Thanks for the feature request.

This is probably not a very popular language feature, available only from a single vendor, and conflicting with standard SQL syntax: https://github.com/jOOQ/jOOQ/issues/1700

As such, I doubt this can be prioritised (possibly ever, given the standard SQL conflict?)

But you can easily add support for this yourself using plain SQL templating: https://www.jooq.org/doc/latest/manual/sql-building/plain-sql-templating/

**Comment by asm0dey** [comment]
Oh, I see, the answer will be the same:
`field("{0} MATCH {1}", Boolean::class.java, field("books_fts"), "<term>")`

**Comment by lukaseder** [comment]
Yeah, or even `condition("{0} MATCH {1}", field("books_fts"), "<term>")`

**Comment by asm0dey** [comment]
For those who is looking for Kotlin-esque solution, here is something more or less working:
```kotlin
fun <X : Record> Table<X>.match(text: Typed<String>): Condition {
    return condition("{0} MATCH {1}", field(unqualifiedName), text)
}
```
